src/drivers/mct/cime_config/buildnml

```diff
@@ -302,44 +302,83 @@ def _create_component_modelio_namelists(case, files):
-                nmlgen.set_value('diri', moddiri)
+        allExist = True
 
-                moddiro = case.get_value('RUNDIR')
-                nmlgen.set_value('diro', moddiro)
+        inst_string = ""
+        inst_index = 1
+        while inst_index <= inst_count:
+            if inst_count > 1:
+                inst_string = '_{:04d}'.format(inst_index)
 
-                logfile = model + inst_string + ".log." + str(lid)
-                nmlgen.set_value('logfile', logfile)
+            modelio_file = model + "_modelio.nml" + inst_string
 
-                # Write output file
-                modelio_file = model + "_modelio.nml" + inst_string
-                nmlgen.write_modelio_file(os.path.join(confdir, modelio_file))
+            mfile = os.path.join(confdir, modelio_file)
 
-                inst_index = inst_index + 1
+            if os.path.exists(mfile):
+                logfile = model + inst_string + ".log." + str(lid)
+                # update the logfile in place
+                for line in fileinput.input(mfile,inplace=1):
+                    if 'logfile' in line:
+                        line = '  logfile = "' + logfile + '"'
+                        print(line)
+                    else:
+                        sys.stdout.write(line)
+            else:
+                print('Could not file ' + str(mfile))
+                allExist = False
+                break
+
+            inst_index = inst_index + 1
+
+        if not allExist:
+            print('Generating modelio namelists for model ' + str(model))
+            with NamelistGenerator(case, definition_file) as nmlgen:
+                config = {}
+                config['component'] = model
+                entries = nmlgen.init_defaults(infiles, config, skip_entry_loop=True)
+
+                inst_string = ""
+                inst_index = 1
+                while inst_index <= inst_count:
+                    # determine instance string
+                    if inst_count > 1:
+                        inst_string = '_{:04d}'.format(inst_index)
+
+                    # set default values
+                    for entry in entries:
+                        nmlgen.add_default(entry)
+
+                    # overwrite defaults
+                    moddiri = case.get_value('EXEROOT') + "/" + model
+                    nmlgen.set_value('diri', moddiri)
+
+                    moddiro = case.get_value('RUNDIR')
+                    nmlgen.set_value('diro', moddiro)
+
+                    logfile = model + inst_string + ".log." + str(lid)
+                    nmlgen.set_value('logfile', logfile)
+
+                    # change for reanalysis: make a simple update to the file if it exists
+                    # Write output file
+                    modelio_file = model + "_modelio.nml" + inst_string
+                    mfile = os.path.join(confdir, modelio_file)
+                   
+                    nmlgen.write_modelio_file(mfile)
```

Comment:
This splitting of the original line into 'mfile = ' and 'nmlgen....'
doesn't seem necessary for the shortcut.  Does it do some other good thing?